internal/cli/atlas/accesslists/create.go

```diff
@@ -80,26 +83,81 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func iPAddress() (string, error) {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP, nil
+	}
+
+	err := fmt.Errorf(currentIPAddressNotFound)
+	return "", err
+}
+
+func needsArg(opts *CreateOpts) bool {
+	// Unless currentIP flag is enabled and type is ip address, args are required.
+	return !(opts.entryType == ipAddress && opts.currentIP)
+}
+
+func (opts *CreateOpts) validateCurrentIPFlag(cmd *cobra.Command, args []string) func() error {
+	if !needsArg(opts) && len(args) > 0 {
+		return func() error {
+			return fmt.Errorf(
+				"please either provide <entry> or use %s to use current IP Address.\n\nUsage: %s",
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
+	}
+
+	if needsArg(opts) && len(args) == 0 {
+		return func() error {
+			return fmt.Errorf(
+				"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				1,
+				len(args),
+				cmd.UseLine(),
+			)
+		}
+	}
+	return func() error { return nil }
+}
+
 // mongocli atlas accessList(s) create <entry> --type cidrBlock|ipAddress|awsSecurityGroup [--comment comment] [--projectId projectId].
 func CreateBuilder() *cobra.Command {
 	opts := &CreateOpts{}
 	cmd := &cobra.Command{
 		Use:   "create <entry>",
 		Short: "Create an IP access list for your project.",
-		Args:  require.ExactArgs(1),
+		Args:  require.MaximumNArgs(1),
 		Annotations: map[string]string{
 			"args":      "entry",
 			"entryDesc": "The IP address, CIDR address, or AWS security group ID of the access list entry to create.",
 		},
+		Example: `  
+		  Create IP address access list with the current IP address. Entry is not needed in this case.
+		  $ mongocli atlas accessList create --currentIP
+		  $ mongocli atlas accessList create --type ipAddress --currentIP
+		`,
```

Comment:
[nit] for backticks spacing and new lines matter (you can see it in the rendered example
```suggestion
		Example: `  Create IP address access list with the current IP address. Entry is not needed in this case.
  $ mongocli atlas accessList create --currentIP
  $ mongocli atlas accessList create --type ipAddress --currentIP
`,
```